src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -717,32 +721,38 @@ int make_crawling_daemon(const po::variables_map &opts)
     ret = rados.get_pool_stats(pool_names, stats);
     if (ret < 0) {
       derr << "error fetching pool stats: " << cpp_strerror(ret) << dendl;
-      return -EINVAL;
+      ret = -EINVAL;
+      goto out;
```

Comment:
This goto is here so that you can call invoke delete state.  Once you've switched it to a unique_ptr, this can go back to being a normal return.